src/ucp/tag/tag_recv.c

```diff
@@ -221,36 +221,19 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_recv_nbx,
     return ret;
 }
 
-UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_msg_recv_nb,
-                 (worker, buffer, count, datatype, message, cb),
-                 ucp_worker_h worker, void *buffer, size_t count,
-                 ucp_datatype_t datatype, ucp_tag_message_h message,
-                 ucp_tag_recv_callback_t cb)
+ucs_status_ptr_t ucp_tag_msg_recv_nb(ucp_worker_h worker, void *buffer, size_t count,
+                                     ucp_datatype_t datatype, ucp_tag_message_h message,
+                                     ucp_tag_recv_callback_t cb)
 {
     ucp_request_param_t param = {
         .op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK |
+                        UCP_OP_ATTR_FIELD_DATATYPE |
                         UCP_OP_ATTR_FLAG_NO_IMM_CMPL,
+        .datatype     = datatype,
```

Comment:
[q] should we initialize `user_data = NULL` here like we did in the test?

Reply:
No, because in test there's a custom generic callback  https://github.com/openucx/ucx/blob/master/test/gtest/ucp/test_ucp_tag.cc#L110 - if a user will pass callback where he expects to get valid user_data but he didn't pass - it's a user error. However there's probably bug here:
https://github.com/openucx/ucx/blob/master/src/ucp/tag/tag_recv.c#L104

Since it sets user data, but checking only callback flag. WDYT?

Reply:
> Since it sets user data, but checking only callback flag. WDYT?

yes, I think this is a bug. So, we also should fix the test to pass USER_DATA, even if it is `NULL`, I guess